src/Lean/Util/TermUnsafe.lean

```diff
@@ -0,0 +1,62 @@
+/-
+Copyright (c) 2021 Microsoft Corporation. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Gabriel Ebner
+-/
+import Lean.Elab.ElabRules
+import Lean.Meta.Closure
+import Lean.Compiler.ImplementedByAttr
+
+/-!
+Defines term syntax to call unsafe functions.
+
+```
+def cool :=
+  unsafe (unsafeCast () : Nat)
+
+#eval cool
+```
+-/
+
+namespace Lean.TermUnsafe
+
+open Lean Meta Elab Term
+
+/-- Construct an auxiliary name based on the current declaration name and the given `hint` base. -/
+def mkAuxName (hint : Name) : TermElabM Name :=
```

Comment:
This feels like a function that wants to be relocated and used more broadly but it can start here

Reply:
I found in Lean.Elab.Term:
```
def Lean.Elab.Term.mkAuxName (suffix : Name) : TermElabM Name := do
  match (← read).declName? with
  | none          => throwError "auxiliary declaration cannot be created when declaration name is not available"
  | some declName => Lean.mkAuxName (declName ++ suffix) 1
```

which is calling another `mkAuxName` in Lean.MonadEnv
```
private partial def mkAuxNameAux (env : Environment) (base : Name) (i : Nat) : Name :=
  let candidate := base.appendIndexAfter i
  if env.contains candidate then
    mkAuxNameAux env base (i+1)
  else
    candidate

def mkAuxName [Monad m] [MonadEnv m] (baseName : Name) (idx : Nat) : m Name := do
  return mkAuxNameAux (← getEnv) baseName idx
```

I am inclined not to spend cycles trying to unify this, just mark this as private (I can mention the existence of the others), and keep moving.